### PR TITLE
PEP 621: Migrate from setup.cfg to pyproject.toml

### PR DESCRIPTION
**Pull request by cclauss**
~Blocked by~
* ~mbarkhau/bumpver#229~

Migrate settings from `setup.cfg` into `pyproject.toml` using [ini2toml](https://pypi.org/project/ini2toml) to do the file conversion and running [pyproject-fmt](https://pypi.org/project/pyproject-fmt) and then [validate-pyproject](https://github.com/abravalheri/validate-pyproject) to validate the results.

### CODE REVIEW

**Review by agjohnson — COMMENTED** [review]
Thanks for this @cclauss! I would feel good about updating our pattern here. The packaging configuration is not something we make use of really, so this is pretty low stakes. This is not a PyPI package and the only place we use this is during release operations.

I think if we update this to work with bumpver it should be okay, I might also be missing something. @humitos am I missing anything?

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+find = { namespaces = false }
+
+[tool.release]
+github_owner = "readthedocs"
+github_repo = "readthedocs.org"
+
+[tool.bumpver]
+current_version = '"11.11.0"'
+version_pattern = '"MAJOR.MINOR.PATCH[TAGNUM]"'
+commit_message = '"Bump version {old_version} -> {new_version}"'
```
[the remark this answers is not in the document]
Comment (cclauss) [reply pyproject.toml:43]: * mbarkhau/bumpver#229

**Comment by humitos** [comment]
> @humitos am I missing anything?

I'm not sure. I think we don't really use setup.cfg as you mentioned, because this is not a package. I understand the only relevant part of this is the `bumpver` section -- which it could be on its own configuration file probably and we can remove the setup.cfg completely. Maybe that make more sense that keep these other files updated?

In general, this work seems to be pretty low priority since we don't really use anything of this 😄

**Comment by agjohnson** [comment]
I don't think we should break out bumpver etc configuration into individual files, I like making everything packages even if we are immediately using them this way.

But if we're upgrading to a new pattern here, we do need bumpver to work.

File: pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (agjohnson) [reply pyproject.toml:43]: Not sure I follow. That seems like another similar change, just on the bumpver package. I meant _we_ need bumpver to work for our repository. We should be able to run the standard bumpver commands and all of the version strings noted in the configuration should update.

**Comment by cclauss** [comment]
My sense is that this pull request is ___blocked by___ mbarkhau/bumpver#229 because it does not make sense for `readthedocs.org` to migrate to `pyproject.toml` until `bumpver` documents how to store its metadata in .toml files.

I have always used [bump-my-version](https://pypi.org/project/bump-my-version) so I have not had this issue before.  

* mbarkhau/bumpver#229

**Comment by agjohnson** [comment]
Bumpver already support this:

https://github.com/mbarkhau/bumpver?tab=readme-ov-file#configuration-setup

That PR does just look like bumpver changing to pyproject.toml for it's own package, not actually supporting any further parsing of pyproject.toml as a library.

The format of the configuration around patterns does indeed seem incorrect in this PR, given the example pyproject.toml.

**Comment by cclauss** [comment]
Thanks for the suggestions.  Configuration fixed.

% `bumpver update --dry --patch`
```diff
INFO    - fetching tags from remote (to turn off use: -n / --no-fetch)
INFO    - Latest version from VCS tag: 2.1.3
INFO    - Working dir version        : 11.11.0
INFO    - Old Version: 11.11.0
INFO    - New Version: 11.11.1

--- docs/conf.py
+++ docs/conf.py
@@ -81,7 +81,7 @@
 
 master_doc = "index"
 copyright = "Read the Docs, Inc & contributors"
-version = "11.11.0"
+version = "11.11.1"
 release = version
 exclude_patterns = ["_build", "shared", "_includes"]
 default_role = "obj"
--- package.json
+++ package.json
@@ -1,6 +1,6 @@
 {
   "name": "readthedocs",
-  "version": "11.11.0",
+  "version": "11.11.1",
   "description": "Read the Docs build dependencies",
   "author": "Read the Docs, Inc <support@readthedocs.com>",
   "scripts": {
--- pyproject.toml
+++ pyproject.toml
@@ -38,7 +38,7 @@
 github_repo = "readthedocs.org"
 
 [tool.bumpver]
-current_version = "11.11.0"
+current_version = "11.11.1"
 version_pattern = "MAJOR.MINOR.PATCH[TAGNUM]"
 commit_message = "Bump version {old_version} -> {new_version}"
 commit = "False"
--- readthedocs/__init__.py
+++ readthedocs/__init__.py
@@ -1,5 +1,5 @@
 """Read the Docs."""
 
 
-__version__ = "11.11.0"
+__version__ = "11.11.1"
 ```

File: setup.cfg [hunk 2]
```diff
@@ -1,49 +0,0 @@
...
-current_version = "11.11.0"
-version_pattern = "MAJOR.MINOR.PATCH[TAGNUM]"
-commit_message = "Bump version {old_version} -> {new_version}"
-commit = False
-tag = False
-push = False
-
-[bumpver:file_patterns]
-setup.cfg = 
-	version = {version}
```
Comment (agjohnson) [inline setup.cfg:41]: This pattern seems to be missing from the pyproject.toml

**Review by agjohnson — COMMENTED** [review]
Nice! That looks better, but I did notice one missing pattern.

I'll let @humitos call it on merging this, as I think if this disrupts anything it will be the release process.

File: setup.cfg (hunk 2 above)
Comment (cclauss) [reply setup.cfg:41]: We deleted `setup.cfg`!

Comment (agjohnson) [reply setup.cfg:41]: This match still exists in pyproject.toml, at https://github.com/readthedocs/readthedocs.org/pull/11726/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R8

This pattern needs to be added to the configuration in pyproject.toml

**Comment by humitos** [comment]
> I don't think we should break out bumpver etc configuration into individual files, I like making everything packages even if we are immediately using them this way.
> But if we're upgrading to a new pattern here, we do need bumpver to work.

That's my point about breaking it into its own file; avoid upgrading something we don't use to a new pattern.

If we don't want to put bumpver into a different file, I'd delete everything that's not used from pyproject.toml and keep only the bumpver configuration there.

However, as I said, this is low priority to me and there aren't too much incentives to upgrade to a new pattern if we are not getting any value from it --mainly because we don't use it.

**Pull request closed without merging**